tests/refs/foreachglob.c

```diff
@@ -48,8 +48,8 @@ static void assert_retrieval(const char *glob, int expected_count)
 
 void test_refs_foreachglob__retrieve_all_refs(void)
 {
-	/* 12 heads (including one packed head) + 1 note + 2 remotes + 7 tags */
-	assert_retrieval("*", 22);
+	/* 12 heads (including one packed head) + 1 note + 2 remotes + 7 tags + 1 blob */
+	assert_retrieval("*", 23);
```

Comment:
Please excuse my ignorance, but I really don't get why there's any changes to either refs::foreachglob or refs::iterator. As far as I can see, both these test suites act on a repository that has been opened via `git_repository_open`. No cloning took place and thus no negotiation via smart protocol. But the code changes you do is only in the smart protocol, so I'm wondering why values in either of those test suites should have changed at all?

Reply:
Good question - I agree.  I think it was a cherry-pick from that other PR, and that introduced a new ref?  I think that this doesn't fit here anymore.

Reply:
Note that I got a green build locally, and was surprised to have it come back red on CI (looking into it). AFAICS this test only adds one of those weird tags-to-non-commits, thus ensuring we correctly ignore it in whatever place we do not (😉) ; hence I'd vote to keep it. I can try adding a "clone this repo" in another testcase to exercise the actual fix to the smart transport.

Reply:
But there is no file added to any test repo, so the reference counts couldn't have changed.

Reply:
Right.  Since `git` looks for the presence of file named `HEAD` to determine whether a directory is a repository or not, our test resources always appear as actual repositories.  So you've probably still got a dirty test directory (with those refs) but you don't know it until you actually look in there.

This has bothered me for some time.

Maybe we should treat `HEAD`s like `.git` files/folders and rename them during test resource copying, to prevent this problem.

Reply:
That was it, sorry about that. GitX sometimes eats `add` entries when building amend indexes, and it slipped out as an untracked file. This should be good now.

Reply:
Nothing to be sorry about! :)